src/js/AppStore.js

```diff
@@ -0,0 +1,194 @@
+                    version: manifest.appVersion,
+                    baseUrl: params.appUrl
+                }), true));
+
+                let addIfExists = (key) => {
+                    if (appDirEntry[key]) {
+                        appProperties[key] = appDirEntry[key];
+                    }
+                }
+
+                addIfExists('nicknames');
+                addIfExists('authToken');
+                addIfExists('transportType');
+                addIfExists('hybridAppPreference');
+
+                BCController.setAppProperties(appProperties);
+            });
+        });
+
+        fileSystemController.sendJSONMessage({
+            method: 'InstallApp',
+            params: {
+                policyAppID: this.state.confirmID,
+                packageUrl: this.state.confirmApp.package.url
+            }
+        });
+
+        this.setState((state, props) => {
+            return {
+                confirmID: null,
+                confirmApp: null
+            }
+        });
+    }
+
+    onSelection(appID) {
+        this.setState((state, props) => {
+            return {
+                confirmID: appID,
+                confirmApp: props.apps.find(app => { return app.policyAppID === appID; })
+            }
+        })
+    }
+
+    componentDidMount() {
+        fetch(flags.AppStoreDirectoryUrl).then((res) => {
+            if (!res.ok) {
+                console.error('could not contact app store server');
+                return [];
+            }
+
+            return res.json();
+        }).then((json) => {
+            store.dispatch(updateAvailableAppStoreApps(json.map((app) => {
+                if ('icon_url' in app) {
+                    app.iconUrl = app.icon_url;
+                    delete app.icon_url;
+                }
+                return app;
+            })));
+        });
+    }
+
+    render() {
+        const themeClass = this.props.theme ? 'dark-theme' : 'light-theme';
+        var modalClass = themeClass + " alertOverlay"
+
+        var pendingInstallApp = this.state.confirmApp ? this.state.confirmApp : {
+            name: null,
+            description: null,
+            iconUrl: null
+        };
+
+        return (
+            <div>
+                <Modal
+                isOpen={!!this.state.confirmID}
+                className="alertModal app-body"
+                overlayClassName={modalClass}
+                contentLabel="Example Modal">
+                    <ConfirmAlert appID={this.state.confirmID} name={pendingInstallApp.name}
+                        description={pendingInstallApp.description} iconUrl={pendingInstallApp.iconUrl}
+                        leftText="Cancel" leftCallback={() => this.cancelInstall()} rightText="Download" rightCallback={() => this.confirmInstall()} />
+                </Modal>
+                <AppHeader icon='store' backLink="/" menuName="APPS"/>
+                {<HScrollMenu data={this.props.apps.map((app) => {
```

Comment:
Please move the map function outside of the return line.

Reply:
done in https://github.com/smartdevicelink/generic_hmi/pull/240/commits/9420e8cd33100df3cba2420ebea12091814adcac